client/ayon_core/hosts/houdini/plugins/publish/collect_files_for_cleaning_up.py

```diff
@@ -0,0 +1,112 @@
+import pyblish.api
+import os
+from ayon_core.pipeline import AYONPyblishPluginMixin
+
+
+class CollectFilesForCleaningUp(pyblish.api.InstancePlugin,
+                                AYONPyblishPluginMixin):
+    """Collect Files For Cleaning Up.
+    
+    This collector collects output files
+    and adds them to file remove list.
+
+    CAUTION:
+        This collector deletes the exported files and
+          deletes the parent folder if it was empty.
+        Artists are free to change the file path in the ROP node.
+    """
+
+    order = pyblish.api.CollectorOrder + 0.2  # it should run after CollectFrames
+
+    hosts = ["houdini"]
+    families = [
+        "camera",
+        "ass",
+        "pointcache",
+        "imagesequence",
+        "mantraifd",
+        "redshiftproxy",
+        "review",
+        "staticMesh",
+        "usd",
+        "vdbcache"
+    ]
+    label = "Collect Files For Cleaning Up"
+
+    def process(self, instance):
+
+        import hou
+
+        node = hou.node(instance.data["instance_node"])
+        filepath = self.get_filepath(node)
+
+        if not filepath:
+            self.log.warning("No filepath value to collect.")
+            return
+    
+        staging_dir, _ = os.path.split(filepath)
+        files = instance.data.get("frames", [])
+        if files: 
+            files = ["{}/{}".format(staging_dir, f) for f in files]
+        else:
+            files = [filepath]
+
+        self.log.debug("Add directories to 'cleanupEmptyDir': {}".format(staging_dir))
+        instance.context.data["cleanupEmptyDirs"].append(staging_dir)
+        
+        self.log.debug("Add files to 'cleanupFullPaths': {}".format(files))
+        instance.context.data["cleanupFullPaths"] += files
+
+    def get_filepath(self, node):
```

Comment:
I think this method works for now but I think it doesn't handle all the cases.
e.g. 
- These Temp folders on Mantra
![image](https://github.com/ynput/ayon-core/assets/20871534/a5942905-cd40-42a5-addc-3e9fda9ac165)

- Render with `split_render` enabled